### PR TITLE
Vet Clinic database: query and update animal table

### PR DESCRIPTION
**Pull request by najibullahjafari**
Here:

- I add a column species of type string to  schema.sql. 
- I added the recommended date to animals table.
- Then I added those queries to data.sql.
- Inside a transaction I updated the animals table by setting the species column to unspecified. 
- Inside a transaction: I updated the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- I updated the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Then I verified, committed and I take a screenshot of that.
- I deleted the columns inside a transaction and all other questions.
-You can see the screenshots here:
Screenshot 2023-08-31 082530
Screenshot 2023-08-31 083325
Screenshot 2023-08-31 083704
Screenshot 2023-08-31 083852
Screenshot 2023-08-31 084105
Screenshot 2023-08-31 084142
Screenshot 2023-08-31 084332
Screenshot 2023-08-31 084924
Screenshot 2023-08-31 085141
Screenshot 2023-08-31 085516

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /* Database schema to keep the structure of entire database. */
 CREATE DATABASE vet_clinic;
 
-CREATE TABLE animals (
+ALTER TABLE animals set
+(
     id SERIAL PRIMARY KEY,
     name VARCHAR(50),
```
Comment (t-yanick) [inline schema.sql:7]: - [ ] Great job working on this milestone. As per project requirements, **Add a column _species_ of type string to your animals table. Modify your schema.sql file.**
   `ALTER TABLE animals ADD species varchar(255);`
Screen Shot 2023-08-31 at 5 43 36 AM

File: queries.sql [hunk 2]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
 
 SELECT * FROM animals WHERE neutered = TRUE;
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
+
+-- Defining the species
+BEGIN;
+UPDATE animals SET species = 'unspecified';
```
Comment (t-yanick) [inline queries.sql:21]: - [ ] Great job working on this milestone. As per project requirements, you were requested to verify that the change was made before rolling back. The line of code below should be on line 19 and Rollback on line 20.
   `SELECT species from animals;`  _verify that change was made_
Screen Shot 2023-08-31 at 5 47 58 AM

File: queries.sql [hunk 3]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
...
+ROLLBACK;
+
```
Comment (t-yanick) [inline queries.sql:25]: - [ ] After the Rollback, you are also expected to show in your code that the change was undone. The line of code below can be used.
`SELECT species from animals;` _-- verify that change was undone_

File: queries.sql [hunk 4]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name = '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = NULL;
+SELECT * FROM animals;
+COMMIT;
+
```
Comment (t-yanick) [inline queries.sql:36]: - [ ] You are supposed to verify that the change persists after the COMMIT. This you can achieve with the line of code below:
`SELECT species from animals;` _-- verify that change persists after commit_

File: queries.sql [hunk 5]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
...
+-- deleting and ROLLBACK
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+
```
Comment (t-yanick) [inline queries.sql:46]: - [ ] Always check after ROLLBACK to ensure the action is undone.
`SELECT COUNT(*) FROM ANIMALS;`

File: queries.sql [hunk 6]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
...
+-- DELETE ANIMALS WITH CONDITIONS
+BEGIN;
+DELETE  FROM animals
+WHERE date_of_birth > '2022-01-01';
```
Comment (t-yanick) [inline queries.sql:50]: - [ ] As per project requirements, after deleting animals born after 2022-01-01, you are expected to create a save point that is missing from your code.
`SAVEPOINT SP1;`
Screen Shot 2023-08-31 at 5 58 51 AM

File: queries.sql [hunk 7]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
...
+SELECT * FROM animals;
+UPDATE animals SET weight_kg = weight_kg *-1;
+ROLLBACK;
```
Comment (t-yanick) [inline queries.sql]: - [ ] This ROLLBACK should be to the savepoint you must have created above.
`ROLLBACK TO SP1;`

File: queries.sql [hunk 8]
```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
...
+UPDATE animals SEt weight_kg= weight_kg * -1 WHERE weight_kg<0;
+
+COMMIT;
+ROLLBACK;
```
Comment (t-yanick) [inline queries.sql]: - [ ] As per project requirements, the last command is the COMMIT. Kindly delete this ROLLBACK command.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @najibullahjafari , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  SQL commands added in the correct files
:heavy_check_mark: Screenshots of queries and transactions added to the PR description
:heavy_check_mark: Use of correct Gitflow

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (najibullahjafari) [reply queries.sql:25]: You know that I verified by SELECT * FROM animals;

File: schema.sql (hunk 1 above)
Comment (rloterh) [reply schema.sql:7]: well resolved! ✔️

File: queries.sql (hunk 2 above)
Comment (rloterh) [reply queries.sql:21]: well resolved! ✔️

File: queries.sql (hunk 3 above)
Comment (rloterh) [reply queries.sql:25]: well resolved! ✔️

File: queries.sql (hunk 4 above)
Comment (rloterh) [reply queries.sql:36]: well resolved! ✔️

File: queries.sql (hunk 5 above)
Comment (rloterh) [reply queries.sql:46]: well resolved! ✔️

File: queries.sql (hunk 6 above)
Comment (rloterh) [reply queries.sql:50]: well resolved! ✔️

File: queries.sql (hunk 7 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

File: queries.sql (hunk 8 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Najibullah,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to your queries!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
 Kindly ensure to add the _Future Features_ section to your ReadMe, which should include 1-3 potential features to be added.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**